libs-scala/crypto/src/main/scala/com/daml/crypto/MacPrototype.scala

```diff
@@ -21,6 +25,9 @@ final class MacPrototype(val algorithm: String) {
       true
     } catch {
       case _: CloneNotSupportedException =>
+        logger.warn(
+          s"${prototype.getClass.getName}.clone() is not supported. It might have implications on performance."
+        )
```

Comment:
Minor: since you chose to use vanilla `slf4j`, Scala string interpolation is executed strictly, regardless of the logging level. This is not really important here I think, since to the best of my understanding you are constructing this objects a few times over the lifetime of the JVM process, but be mindful of building strings as part of logging depending on the logging framework you use. `scala-logging` and `contextualized-logging` use string interpolation lazily. When using vanilla `slf4j` or other Java logging frameworks you should use built-in support for string formatting (e.g. https://www.slf4j.org/apidocs/org/slf4j/Logger.html#debug-java.lang.String-java.lang.Object...-)

Reply:
Thanks for the reminder, it's a good point.